### PR TITLE
Compose customer-facing invoice lines from booking semantics

### LINKED ISSUE #4936: Compose customer-facing invoice lines from commercial booking semantics, not raw booking-item rows
## Problem

Invoice generation currently maps persisted booking-item rows 1:1 to invoice lines. Those rows are useful for booking allocation and pricing, but they are not always meaningful customer-facing commercial concepts.

For example, a supplement-priced package can persist:

- an `Adult` per-person row,
- a priced `SGL` room supplement row, and
- a zero-value `DBL` room allocation row.

Finance uses the product snapshot as the preferred display name for every row. Consequently, several invoice lines can have the same product/date description while carrying different quantities and prices, with no indication of whether a line represents travelers, rooms, or a supplement. A downstream accounting integration cannot repair this because it correctly forwards the persisted invoice lines unchanged.

Recent booking fixes improved how new booking items are priced and modeled, but did not change this invoice composition behavior.

## Expected behavior

Invoices created from bookings should express the commercial purchase in terms a traveler and agency can understand, while preserving the exact subtotal, tax, currency, and total.

This must be automatic and convention-driven. Agencies should not have to rename booking items, manually consolidate lines, or configure invoice mappings.

For the normal case where a package and all of its priced components share the same fiscal treatment, Finance should create one customer-facing line using the product name, and include the purchased option/room combination in the line details. For example:

> Package name — 3 travelers — 1 × SGL + 1 × DBL

The line quantity may be `1` for the composed package and its price should be the exact combined booking total. When components have incompatible tax rates, currencies, or other legally relevant fiscal attributes, Finance should split them into the minimum number of clearly named lines required to preserve those attributes.

The composed text must use the invoice/document locale. This includes the localized product name and localized traveler, option, and room labels. If the requested locale is unavailable for a particular value, use the product's available/default name through a deterministic fallback rather than requiring agency intervention. Persist the resolved text on the invoice at creation time so later translation or product edits cannot alter an issued document.

- Allocation-only, zero-value rows should not become fiscal lines.
- Distinct priced concepts should have distinct descriptions (for example, base traveler price versus single-room supplement).
- Equivalent rows may be grouped only when their service identity, unit price, tax treatment, currency, and other fiscal attributes make grouping safe.
- All-in room pricing must not invent a redundant traveler-price proxy line.
- The persisted invoice lines must remain the source of truth for downstream accounting integrations.

## Reproduction

1. Create a supplement-priced package with an adult per-person price, a priced single-room supplement, and a double-room allocation option.
2. Create a booking with multiple adults and a mixture of room allocations.
3. Generate an invoice from the booking.
4. Observe that raw booking-item rows become invoice lines and may all use the same product/date description, including allocation-only rows.

## Proposed boundary

Introduce a customer-facing invoice-line composer in Finance before invoice persistence, rather than changing downstream accounting adapters. The composer will need enough booking-unit/option semantics to distinguish base traveler prices, room supplements, and allocation-only rows.

The public invoice-from-booking API can remain unchanged.

## TDD implementation plan

1. Add a failing Finance integration test for a supplement-priced booking containing `Adult` quantity 3, a priced `SGL` row, and a zero-value `DBL` allocation row. Assert intelligible, non-duplicate fiscal descriptions; exclusion of the zero-value allocation row; and unchanged subtotal/tax/total.
2. Implement the smallest invoice-line composition rule that satisfies the supplement-pricing case, using booking unit/option semantics rather than display-name heuristics. When fiscal attributes are compatible, emit one product line with the traveler count and room/option combination in its details.
3. Add a failing test for all-in room pricing. Assert that it describes the purchased room/service without emitting a redundant `Adult` proxy line.
4. Add failing tests for multiple priced components sharing the same product/date snapshot. Keep them distinguishable, or group only when service identity, unit price, tax, and currency are compatible.
5. Add a contract test proving that the accounting integration event/payload exactly reflects the persisted composed invoice lines and does no independent regrouping or relabeling.
6. Add locale tests proving that a Romanian invoice uses the Romanian product and component labels, falls back deterministically when only an English/default value exists, and snapshots the resolved description at invoice creation.
7. Add a legacy regression fixture shaped like an already-issued invoice whose raw rows have identical product descriptions. Document and test the supported recovery path (void/credit and reissue); never silently rewrite an issued fiscal document.

## Acceptance criteria

- A newly generated invoice never exposes zero-value allocation-only rows.
- A reader can distinguish traveler/base pricing from room supplements or other priced components.
- Invoice totals, tax, and currency remain exact.
- Safe grouping rules are deterministic and covered by tests.
- Downstream integrations remain pass-through consumers of persisted invoice lines.
- Customer-facing line text uses the invoice locale with a deterministic fallback, and remains immutable after issuance.
- Already-issued documents are not mutated; recovery requires an explicit fiscal correction workflow.

### PR DESCRIPTION
**Pull request by mihaipxm**
Closes #4936

## What changed

- compose compatible package booking components into the minimum customer-facing fiscal lines before invoice persistence
- use persisted product/option/unit semantics, including allocation-only room selections, without turning zero-value allocations into fiscal lines
- split components when service identity, dates, currency, or complete tax treatment is incompatible
- resolve product and component copy in the document locale with deterministic snapshot/default fallback, then persist the resolved description
- preserve composed invoice lines as the source for `invoice.issued` accounting payloads
- document and test explicit void-and-reissue recovery for legacy issued documents; existing fiscal lines are never rewritten
- expose the optional document locale on invoice-from-booking and regenerate the Finance admin OpenAPI document

## Verification

- `vitest run src/invoice-line-composer.test.ts tests/unit/validation-billing.test.ts` (15 passed)
- Finance route integration file transforms successfully (152 database tests registered; locally skipped because the Docker database engine was unavailable)
- Finance contracts typecheck: clean
- Biome check on all changed TypeScript files: clean
- OpenAPI checker unit tests: 16 passed
- Finance OpenAPI generator: clean; generated admin contract committed

The full Finance typecheck was attempted locally, but host-wide swap pressure caused TypeScript to remain in paging with no diagnostics; CI is the clean-install authority for that lane.

### CODE REVIEW

File: packages/finance/src/service-invoice-from-booking.ts [hunk 1]
```diff
@@ -249,6 +267,62 @@ export const financeInvoiceFromBookingService = {
...
     }
 
+    const invoiceLocale =
+      data.language?.trim() ||
+      booking.contactPreferredLanguage?.trim() ||
+      booking.communicationLanguage?.trim() ||
+      "en"
+    const lineContent =
+      shouldUseBookingItems && invoiceItems.length > 0
+        ? await resolveInvoiceLineContent(db, bookingData, invoiceLocale)
```
[the remark this answers is not in the document]
Comment (mihaipxm) [reply packages/finance/src/service-invoice-from-booking.ts]: Resolved in 88c2148387. Approval preview/revalidation and invoice creation now share , including locale resolution, catalog translations, selections, fiscal grouping, and composed descriptions. Atomic revalidation also locks the projection rows it reads. Added a DB regression proving a translation change invalidates the approved fingerprint before any invoice is created.

File: packages/finance/src/service-invoice-from-booking.ts [hunk 2]
```diff
@@ -155,6 +158,21 @@ async function resolveInvoiceNumberForBooking(
...
+function invoiceItemTaxSignature(taxes: Array<typeof bookingItemTaxLines.$inferSelect>) {
+  return JSON.stringify(
+    taxes.map((tax) => [
+      tax.scope,
+      tax.code,
+      tax.name,
+      tax.jurisdiction,
+      tax.currency,
+      tax.rateBasisPoints,
+      tax.includedInPrice,
```
[the remark this answers is not in the document]
Comment (mihaipxm) [reply packages/finance/src/service-invoice-from-booking.ts]: Resolved in 88c2148387. Candidate fiscal groups are now partitioned unless provider-side aggregate tax recomputation exactly equals the sum of persisted item taxes. Added pure and DB/API regressions for two 3-cent items with 19% VAT (1 cent each): they remain two lines, persist taxCents=2, and emit two 19% provider lines.

File: packages/finance/src/invoice-line-composer.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+  const groups = new Map<string, FiscalGroup>()
+  for (const item of pricedItems) {
+    const key = fiscalGroupKey(item)
+    const group = groups.get(key)
+    if (group) group.items.push(item)
+    else groups.set(key, { key, items: [item] })
+  }
+
+  const splitsFiscalTreatment = groups.size > 1
```
[the remark this answers is not in the document]
Comment (mihaipxm) [reply packages/finance/src/invoice-line-composer.ts]: Resolved in 88c2148387. Split detection is now scoped to other fiscal groups for the same product/option/service window, so an unrelated second service no longer suppresses traveler count on an all-in package. Added a focused regression for that case.
